### ISSUE TITLE
Issue running on Archlinux

### ISSUE THREAD

**Issue by qjerome** [Vector35/binaryninja-api#780]
Hi,

I have just bought BinNinja and I am not able to run it on my distrib (Archlinux), even though I followed the recommendations in: https://docs.binary.ninja/guide/troubleshooting/#linux

I have an issue running BinNinja on Archlinux. I get a `SEGFAULT` in libssl.so, only when I do the libpython symlink as explained in the Archlinux troubleshooting part of the guide:
 * Install python2 from the official repositories (sudo pacman -S python2) and create a symlink: sudo ln -s /usr/lib/libpython2.7.so.1.0 /usr/lib/libpython2.7.so.1

Output after creating the symlink:
```
LD_PRELOAD=libcurl.so.3 ./binaryninja --debug --stderr-log        

./binaryninja: /usr/lib/libcurl.so.3: no version information available (required by /home/quentin/.local/opt/binaryninja/libbinaryninjacore.so.1)
[1]    3256 segmentation fault (core dumped)  LD_PRELOAD=libcurl.so.3 ./binaryninja --debug --stderr-log
```
Syslog
```
binaryninja[3256]: segfault at 140 ip 000062f69692c500 sp 00007b449ce761b8 error 6 in libssl.so.1.1[62f6968fe000+61000]
```

Please note that it starts properly without this symlink and I actually don't know if I am going to miss alot of features without this library being loaded !

I don't necessarily expect you to find the solution to this problem since it looks rather specific, it is just to let you know that some users using Archlinux may have issues running BinNinja.

Thank you

**Comment by metafex** [comment]
Have you tried the pkgbuild in aur? (https://aur.archlinux.org/packages/binaryninja-personal)

It's not yet updated to 1.1 but will be tomorrow.

**Comment by qjerome** [comment]
I have found another solution that fixes everything up (which I find a little cleaner than doing a symlink in /usr/lib with sudo rights, as explained in your )

I saw there was a version of libssl.so in ./plugin of the archive, so I decided to try to use it since the SEGFAULT came from the libssl installed in Arch.

So the fix is the following:
1. Fix cURL issue:  ln -s /usr/lib/libcurl-compat.so.4.4.0 $BINNINJA_INSTALL/plugins/libcurl.so.4
2. Fix Python issue: ln -s /usr/lib/libpython2.7.so.1.0 $BINNINJA_INSTALL/plugins//usr/lib/libpython2.7.so.1
3. Fix loader so that it takes our library fixes into account + the libssl.so packaged with BinaryNinja : LD_LIBRARY_PATH=$BINNINJA_INSTALL/plugins ./binaryninja

Everything done in user mode and is working for me
